recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-bsp/u-boot/files/0001-Enable-everything-needed-source-command-to-work.patch

```diff
@@ -0,0 +1,52 @@
+From e94f84b9febb4ff4e41f758a58434fb0641b7642 Mon Sep 17 00:00:00 2001
+From: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
+Date: Fri, 2 Feb 2018 21:42:51 +0200
+Subject: [PATCH] Enable everything needed source command to work
+
+Fix for the cmd_source was taken here:
+https://lists.denx.de/pipermail/u-boot/2015-July/221152.html
+
+Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
+---
+ common/cmd_source.c                | 4 ++--
+ include/configs/rcar-gen3-common.h | 4 ++++
+ 2 files changed, 6 insertions(+), 2 deletions(-)
+
+diff --git a/common/cmd_source.c b/common/cmd_source.c
+index 6881bc9..5f550a5 100644
+--- a/common/cmd_source.c
++++ b/common/cmd_source.c
+@@ -32,7 +32,7 @@ source (ulong addr, const char *fit_uname)
+ #if defined(CONFIG_IMAGE_FORMAT_LEGACY)
+ 	const image_header_t *hdr;
+ #endif
+-	ulong		*data;
++	u32		*data;
+ 	int		verify;
+ 	void *buf;
+ #if defined(CONFIG_FIT)
+@@ -73,7 +73,7 @@ source (ulong addr, const char *fit_uname)
+ 		}
+ 
+ 		/* get length of script */
+-		data = (ulong *)image_get_data (hdr);
++		data = (u32 *)image_get_data (hdr);
+ 
+ 		if ((len = uimage_to_cpu (*data)) == 0) {
+ 			puts ("Empty Script\n");
+diff --git a/include/configs/rcar-gen3-common.h b/include/configs/rcar-gen3-common.h
+index 24ce2de..0bef2e7 100644
+--- a/include/configs/rcar-gen3-common.h
++++ b/include/configs/rcar-gen3-common.h
+@@ -125,4 +125,8 @@
+ #define CONFIG_SYS_MALLOC_LEN		(1 * 1024 * 1024)
+ #define CONFIG_SYS_BOOTMAPSZ		(8 * 1024 * 1024)
+ 
++#define CONFIG_CMD_SOURCE
++#define CONFIG_CMD_ECHO
++#define CONFIG_IMAGE_FORMAT_LEGACY /* enable also legacy image format */
++
```

Comment:
This is rather a build configuration change. Please separate it from fix.

Reply:
done